nap/engine.py

```diff
@@ -594,3 +594,51 @@ def get_full_url(self, uri):
         full_url = '{}/{}'.format(root_url.rstrip('/'), uri.lstrip('/'))
 
         return full_url
+
+
+class OpaqueFilterResourceEngine(ResourceEngine):
+    def filter(self, skip_cache=True, **lookup_vars):
+        """
+        Accesses the first URL set as a collections URL with no additional
+        parameters passed. Returns a list of current ResourceModel objects
+
+        Drops support for collection_field from ResourceEngine.filter
+
+        :param skip_cache: If true don't cache results (defaults to true
+        for backwards compatibility) and don't check cache for an existing
+        value for this request
```

Comment:
I'm curious if we want to change this default since we don't need to worry about backward compatibility for the new engine.

Reply:
Good thinking; I don't see why not. I'll make that change.